addons/rating/views/rating_view.xml

```diff
@@ -24,7 +24,9 @@
                     <sheet>
                         <group>
                             <group>
-                                <field name="res_name"/>
+                                <field name="res_model" invisible="1" />
+                                <field name="res_id" invisible="1" />
+                                <field name="res_name" widget="link_record" options="{'res_model_field': 'res_model', 'res_id_field': 'res_id'}" />
```

Comment:
I bet an action (in python) that return the good res_model/res_id can make it. Is a new widget really required ?